mods/pvp/hud_score/init.lua

```diff
@@ -0,0 +1,103 @@
+hud_score = {}
+local hud = hudkit()
+
+local players = {}
+local duration = 5
+local max = 6
+
+local function update(name)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		return
+	end
+
+	-- Handle all elements marked for deletion
+	-- and rebuild table
+	local temp = {}
+	for _, def in ipairs(players[name]) do
+		if def.delete then
+			if hud:exists(player, def.name) then
+				hud:remove(player, def.name)
+			end
+		else
+			table.insert(temp, def)
+		end
+	end
+
+	for i, def in ipairs(temp) do
+		-- If not the top-most element, prefix with "+ "
+		local text = def.value
+		if i > 1 then
+			text = "+ " .. text
+		end
+
+		if hud:exists(player, def.name) then
+			hud:change(player, def.name, "text", text)
+			hud:change(player, def.name, "offset", {x = 0, y = i * 20})
+		else
+			hud:add(player, def.name, {
+				hud_elem_type = "text",
+				alignment = {x = 0, y = 0},
+				position = {x = 0.5, y = 0.6},
+				offset = {x = 0, y = i * 20},
+				number = def.color,
+				text = text
+			})
+		end
+	end
+	players[name] = temp
+end
+
+function hud_score.new(name, def)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		return
+	end
+
+	-- Verify HUD score element def
+	if not name or not def or not type(def) == "table" or
```

Comment:
http://www.lua.org/manual/5.1/manual.html#2.5.6
`not type(def)` will be `false` and `false == "table"` is `false`. So `type(def) == "table"` is always `false` because `type` gives back strings and `not` has a higher precedence than `==`.
Use `~=`!
```suggestion
	if not name or not def or type(def) ~= "table" or
```